homeassistant/helpers/entity_platform.py

```diff
@@ -646,7 +614,14 @@ async def _async_add_entity(  # noqa: C901
                     return
 
             if self.config_entry and (device_info := entity.device_info):
-                device = self._async_process_device_info(device_info)
+                try:
+                    device = dev_reg.async_get(self.hass).async_get_or_create(
+                        config_entry_id=self.config_entry.entry_id,
+                        **device_info,
```

Comment:
`**` ends up biting us from time to time when things are made for other versions and keys change. Should we be worried here ?

Reply:
I really think it's OK here. In a follow-up, we should move the definition of `DeviceInfo` to device_registry.py, and once we upgrade to Python 3.12, we should let `async_get_or_create` accept kwargs types as `DeviceInfo`